cpp/lib/src/SyndicationFormat.cc

```diff
@@ -324,15 +324,19 @@ std::unique_ptr<SyndicationFormat::Item> Atom::getNextItem() {
     std::string data;
     while (xml_parser_->getNext(&type, &attrib_map, &data)) {
         if (type == SimpleXmlParser<StringDataSource>::CLOSING_TAG and data == item_tag_)
-            return std::unique_ptr<SyndicationFormat::Item>(new Item(title, summary, id, link, updated));
+            return std::unique_ptr<SyndicationFormat::Item>(new Item(title, summary, link, id, updated));
         else if (type == SimpleXmlParser<StringDataSource>::END_OF_DOCUMENT)
             return nullptr;
         else if (type == SimpleXmlParser<StringDataSource>::OPENING_TAG and data == "title")
             title = ExtractText(xml_parser_, "title");
         else if (type == SimpleXmlParser<StringDataSource>::OPENING_TAG and data == "summary")
             summary = ExtractText(xml_parser_, "summary");
-        else if (type == SimpleXmlParser<StringDataSource>::OPENING_TAG and data == "link")
+        else if (type == SimpleXmlParser<StringDataSource>::OPENING_TAG and data == "link") {
+            // TODO do we need to do this for the other feed types as well?
             link = ExtractText(xml_parser_, "link");
+            if (link.empty() and attrib_map.count("href"))
```

Comment:
We never use integers as booleans, please rewrite this either as "attrib_map.count("href") == 1" or, preferred as "attrib_map.find("href") != attrib_map.cend()".